frontend/src/App.tsx

```diff
@@ -102,15 +94,48 @@ function AppContent() {
     try {
       await configApi.saveConfig(updatedConfig);
       setShowPrivacyOptIn(false);
-      // Now show GitHub login after privacy choice is made
-      const notAuthenticated =
-        !updatedConfig.github?.username || !updatedConfig.github?.token;
-      setShowGitHubLogin(notAuthenticated);
     } catch (err) {
       console.error('Error saving config:', err);
     }
   };
 
+  const handleGitHubLoginComplete = async () => {
+    if (!config) return;
+
+    setShowGitHubLogin(false);
+
+    try {
+      // Add a delay to ensure backend has time to save the GitHub token
+      // (if user authenticated) before refreshing the config
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // Refresh the config to get the latest GitHub authentication state
+      const latestConfig = await configApi.getConfig();
+      updateConfig(latestConfig);
+
+      // If user skipped (no GitHub token), we need to manually set the acknowledgment
+      if (!latestConfig.github?.token) {
+        const updatedConfig = {
+          ...latestConfig,
+          github_login_acknowledged: true,
+        };
+        updateConfig(updatedConfig);
+        await configApi.saveConfig(updatedConfig);
+      }
```

Comment:
shouldn't we just update config with github_login_acknowledged as soon as the user either closes the popup or logs in? Also, we could have some function in useConfig() to just refetch config instead of requesting it here and updating it twice